src/playlist-selectors.js

```diff
@@ -124,27 +146,32 @@ const simpleSelector = function(master, bandwidth, width, height) {
   let resolutionPlusOneSmallest = [];
   let resolutionBestVariantList = [];
 
-  stableSort(sortedPlaylists, comparePlaylistBandwidth);
+  stableSort(sortedPlaylists, comparePlaylistBandwidth.bind(null, useAverageBandwidth));
 
   // filter out any playlists that have been excluded due to
   // incompatible configurations or playback errors
   sortedPlaylists = sortedPlaylists.filter(Playlist.isEnabled);
   // filter out any variant that has greater effective bitrate
   // than the current estimated bandwidth
   bandwidthPlaylists = sortedPlaylists.filter(function(elem) {
-    return elem.attributes &&
-           elem.attributes.BANDWIDTH &&
-           elem.attributes.BANDWIDTH * Config.BANDWIDTH_VARIANCE < bandwidth;
+    const elemBandwidth = getBandwidth(elem, useAverageBandwidth);
```

Comment:
`In practice, most m3u8s should exhibit all or nothing, either AVERAGE-BANDWIDTH is present on all playlists, or it is present on none.`
I agree with this, and a "best available information" approach for this case seems like it will still give desired results.

`it is possible for a playlist to have a higher BANDWIDTH but a lower AVERAGE-BANDWIDTH than another playlist`
I'm not entirely convinced this means we want to use the average when sorting. If I'm not mistaken, the goal of the playlist selector is to select the highest bandwidth+resolution we can support with the current system bandwidth. If a playlist with a lower BANDWIDTH has a higher AVERAGE than others, then it will be considered to have a higher bandwidth in sorting. This could make the player select the playlist with a lower overall bandwidth but higher average. Maybe this is desired, but from my understanding it is not. 

Redundant streams with differing averages still concerns me, but I haven't seen any real world examples of redundant streams with average-bandwidth to know whether they are likely to differ or not.